server/modules/activityLog/repository/activityLogRepository.js

```diff
@@ -45,13 +45,17 @@ export const fetch = async (surveyId, activityTypes = null, offset = 0, limit =
       )
     SELECT
       l.*,
-      u.name AS user_name
+      u.name AS user_name,
+      r.uuid IS NULL AS record_deleted
```

Comment:
I wouldn't make this assumption here. the record is deleted if the activity log is related to a record and if record uuid is null. I would simply call this record_uuid and check if it's deleted in the isItemDeletedFn

Reply:
ok